sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowTablesExtendedExec.scala

```diff
@@ -0,0 +1,201 @@
+        val table = catalog.loadTable(tableIdent)
+        val information = getTableDetails(catalog.name, tableIdent, table)
+        rows += toCatalystRow(tableIdent.namespace().quoted, tableIdent.name(), false,
+          s"$information\n")
+        }
+      }
+
+    // fetch views, includes: view, global temp view, local temp view
+    val sessionCatalog = session.sessionState.catalog
+    val (namespaceExists, db) = namespace match {
+      case Seq(db) =>
+        (sessionCatalog.databaseExists(db), Some(db))
+      case _ =>
+        (false, None)
+    }
+    if (namespaceExists) {
+      val views = sessionCatalog.listViews(db.get, pattern)
+      views.map { viewIdent =>
+        val tableName = viewIdent.table
+        val isTemp = sessionCatalog.isTempView(viewIdent)
+        val view = sessionCatalog.getTempViewOrPermanentTableMetadata(viewIdent)
+        val information = view.simpleString
+        rows += toCatalystRow(db.get, tableName, isTemp, s"$information\n")
+      }
+    }
+
+    rows.toSeq
+  }
+
+  private def getTableDetails(
+      catalogName: String,
+      identifier: Identifier,
+      table: Table): String = {
+    val results = new mutable.LinkedHashMap[String, String]()
+
+    results.put("Catalog", catalogName)
+
+    if (!identifier.namespace().isEmpty) {
+      results.put("Namespace", identifier.namespace().quoted)
+    }
+    results.put("Table", identifier.name())
+    val tableType = if (table.properties().containsKey(TableCatalog.PROP_EXTERNAL)) {
+      CatalogTableType.EXTERNAL
+    } else {
+      CatalogTableType.MANAGED
+    }
+    results.put("Type", tableType.name)
+
+    CatalogV2Util.TABLE_RESERVED_PROPERTIES
+      .filterNot(_ == TableCatalog.PROP_EXTERNAL)
+      .foreach(propKey => {
+        if (table.properties.containsKey(propKey)) {
+          results.put(propKey.capitalize, table.properties.get(propKey))
+        }
+      })
+
+    val properties =
+      conf.redactOptions(table.properties.asScala.toMap).toList
+        .filter(kv => !CatalogV2Util.TABLE_RESERVED_PROPERTIES.contains(kv._1))
+        .sortBy(_._1).map {
+        case (key, value) => key + "=" + value
+      }.mkString("[", ",", "]")
+    if (table.properties().isEmpty) {
+      results.put("Table Properties", properties.mkString("[", ", ", "]"))
+    }
+
+    // Partition Provider & Partition Columns
+    var partitionColumns = new StructType()
+    if (table.supportsPartitions && table.asPartitionable.partitionSchema().nonEmpty) {
+      partitionColumns = table.asPartitionable.partitionSchema()
+      results.put("Partition Provider", "Catalog")
+      results.put("Partition Columns", table.asPartitionable.partitionSchema().map(
+        field => quoteIdentifier(field.name)).mkString("[", ", ", "]"))
+    }
+
+    if (table.schema().nonEmpty) {
+      val dataColumns = table.schema().filterNot(partitionColumns.contains)
+      results.put("Schema", StructType(dataColumns ++ partitionColumns).treeString)
```

Comment:
shouldn't we use `Table#schema`?

Reply:
My first version used `Table#schema` directly, but during UT testing, it was found that the schema field information partition field was not displayed at the end, which is different from the result of the `V1 command`. Do we want to maintain consistency? Or this order is actually not important.

Reply:
Yes. Shall we output the partition field even if it is empty?

Reply:
I think we should respect the actual table schema. We can add one more abstraction in the base test suite to allow this difference.

Reply:
@panbingkun Seems this is missed. We should use the original table schema here.

Reply:
Done